### PR TITLE
fix #17490

### LINKED ISSUE #17490: js: parseJson doesn't work properly with large numbers
### Example
```nim
import std/json

let a = "18446744073709551615"
let b = a.parseJson
echo b.kind
let c = $b
echo b # fails
```

### Current Output
```
JString
/Users/timothee/git_clone/nim/timn/build/nimcache/t12061.js:171
                  var code_33556847 = c_33556834.toString(16);
                                                 ^

TypeError: Cannot read property 'toString' of undefined
    at toJSStr (/Users/timothee/git_clone/nim/timn/build/nimcache/t12061.js:171:50)
    at rawEcho (/Users/timothee/git_clone/nim/timn/build/nimcache/t12061.js:481:16)
    at Object.<anonymous> (/Users/timothee/git_clone/nim/timn/build/nimcache/t12061.js:5599:1)
    at Module._compile (node:internal/modules/cjs/loader:1092:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1121:10)
    at Module.load (node:internal/modules/cjs/loader:972:32)
    at Function.Module._load (node:internal/modules/cjs/loader:813:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
    at node:internal/main/run_main_module:17:47
```

### Expected Output
18446744073709551615

### Additional Information
1.5.1 28221e8a33e111d8653c4ff21fad47a7ec6ef54d
found while working on https://github.com/nim-lang/Nim/pull/17469

in js:
```
a="18446744073709551615"
b = JSON.parse(a)
console.log(typeof(b), b)

prints:
number 18446744073709552000
```

the question is what to do here:
option 1: do as js (and give approximate result)
option 2: raise an exception
option 3: check that the number represents a safe js integer, and if not, use `newJRawNumber`

## note
1.4.4 gives:
JInt
18446744073709552000

### PR DESCRIPTION
**Pull request by ringabout**
fix #17490

### CODE REVIEW

File: tests/stdlib/tjson.nim [hunk 1]
```diff
@@ -300,3 +300,14 @@ block: # bug #17383
   when not defined(js):
     testRoundtrip(int64.high): "9223372036854775807"
     testRoundtrip(uint64.high): "18446744073709551615"
+
+
```
Comment (timotheecour) [inline tests/stdlib/tjson.nim:304]: optional: can you also add a test in appropriate place in tjsonutils?

File: tests/stdlib/tjson.nim [hunk 2]
```diff
@@ -300,3 +300,14 @@ block: # bug #17383
...
+block:
+  let a = "18446744073709551615"
+  let b = a.parseJson
+  doAssert b.kind == JString
+  let c = $b
+  when defined(js):
+    doAssert c == "18446744073709552000"
+  else:
+    doAssert c == "18446744073709551615"
```
Comment (timotheecour) [inline tests/stdlib/tjson.nim:313]: can't we do option 3 so that `c == "18446744073709551615"` even in js?

Comment (ringabout) [reply tests/stdlib/tjson.nim:313]: not necessary to be in this PR, the PR fixes the regression.

Comment (timotheecour) [reply tests/stdlib/tjson.nim:313]: ok

File: lib/pure/json.nim [hunk 3]
```diff
@@ -946,13 +947,13 @@ when defined(js):
     else: assert false
 
   proc len(x: JSObject): int =
-    assert x.getVarType == JArray
```
Comment (ringabout) [inline lib/pure/json.nim:949]: It is unnecessary for its usage here.

**Review by timotheecour — APPROVED** [review]

**Pull request merged**